networkit/cpp/independentset/Luby.cpp

```diff
@@ -42,28 +44,33 @@ std::vector<bool> Luby::run(const Graph &G) {
     while (!empty()) {
         // choose set S - weighted choice of active nodes with probability $1 / 2 \omega(v)$
         std::vector<bool> S(G.numberOfNodes(), false);
+
         G.parallelForNodes([&](node u) {
             if (V[u]) {
                 if (Aux::Random::probability() < nodeProbability(u)) {
                     S[u] = true; // add node to S
                 }
             }
         });
+
         // remove non-independent nodes from S to get S'
-        G.parallelForEdges([&](node u, node v) {
-            if (u != v) {           // exclude self-loops
-                if (S[u] && S[v]) { // u and v are not independent (note: S is subset of V')
-                    // remove node with smaller degree
-                    edgeweight wu = weightedDegree(u);
-                    edgeweight wv = weightedDegree(v);
-                    if (wu > wv) {
-                        S[v] = false;
-                    } else if (wv > wu) {
-                        S[u] = false;
-                    } else {          // tie
-                        S[v] = false; // arbitrary decision
+        G.parallelForNodes([&](node u) {
+            if (S[u]) {
+                G.forNeighborsOf(u, [&](node v) {
+                    if (u < v) { // exclude self-loops and enforce strict order
+#pragma omp critical
+                        {
+                            // remove node with smaller degree from S
+                            if (S[u] && S[v]) {
```

Comment:
You can avoid two indentation levels:
```suggestion
                    // exclude self-loops and enforce strict order
                    if (u >= v ) return; 
#pragma omp critical
                        {
                            // remove node with smaller degree from S
                            if (!(S[u] && S[v])) return;
```

Reply:
This suggestion is not cleanly implementable: Returns are not allowed in a 'pragma omp critical' section, and using any 'omp cancellation' turns into error: ‘#pragma omp cancel’ may only be used in compound statements.